### PR TITLE
AQCU-1052 first pass fix for extremes report

### PR DESCRIPTION
**Pull request by mhines-usgs**
error seen, when primary and related upchain number of values mismatch an internal error occurred
now, it merges the two datasets into one dataframe and assigns NAs to values missing in one or the other.

### CODE REVIEW

File: R/extremes-data.R [hunk 1]
```diff
@@ -210,7 +212,27 @@ createDataRows <-
         {
           relatedValue <- relatedSet$value
         }
+        
+        if (!isEmptyOrBlank(x$relatedPrimary)) {
+          if (nrow(x$relatedPrimary) > nrow(x$points) ||
+              nrow(x$points) > nrow(x$relatedPrimary)) {
```
Comment (dpattermann-usgs) [inline R/extremes-data.R]: Could these two if statements be simplified to nrow(x$related{Primary or Upchain}) != nrow(x$points) ?

Comment (mhines-usgs) [reply R/extremes-data.R]: is that a thing?! i'm actually writing a whole separate function now to reduce the duplicity of these two sets of statements as well and will try that..

Comment (dpattermann-usgs) [reply R/extremes-data.R]: Ha, I just mean replace each of those statements, it won't just be one statement.

Comment (dpattermann-usgs) [reply R/extremes-data.R]: Like (if(nrow(x$relatedPrimary) != nrow(x$points){ blah }
       if(nrow(x$relatedUpchain) != nrow(x$points)){ blah }
Sorry for the confusion, I don't have any secret R magic to get rid of the duplication off the top of my head.

**Review by zmoore-usgs — CHANGES_REQUESTED** [review]
The approach looks good to me, though I'd suggest instead of giving each row the footnote string perhaps "footnote" should just hold TRUE or FALSE and then instead of including the if/else in the for loop in extremesTable I think you could just do return(list(toRet=toRet, footnote=any(dataRows$footnote))

Then in the RMD file add the actual footnote text if footnote=true, similar to how the other notes are being added about the '*' characters

**Comment by mhines-usgs** [comment]
thanks @zmoore-usgs now that I got all the tests working again, I'll give that a try.

**Comment by zmoore-usgs** [comment]
I took this ticket from @mhines-usgs after she left and made a few tweaks. New PR: https://github.com/USGS-R/repgen/pull/637

**Pull request closed without merging**